ra/ra.go

```diff
@@ -566,108 +566,115 @@ func (ra *RegistrationAuthorityImpl) checkNewOrdersPerAccountLimit(ctx context.C
+	if req == nil || req.Authz.Identifier == "" || req.RegID == 0 {
+		return nil, errIncompleteGRPCRequest
+	}
+
+	// Create ACMEIdentifier. Assume Type DNS.
+	acmeIdentifier := identifier.ACMEIdentifier{
+		Type:  identifier.DNS,
+		Value: strings.ToLower(req.Authz.Identifier),
+	}
 
 	// Check that the identifier is present and appropriate
-	if err := ra.PA.WillingToIssue(identifier); err != nil {
-		return core.Authorization{}, err
+	if err := ra.PA.WillingToIssue(acmeIdentifier); err != nil {
+		return nil, err
 	}
 
-	if err := ra.checkPendingAuthorizationLimit(ctx, regID); err != nil {
-		return core.Authorization{}, err
+	if err := ra.checkPendingAuthorizationLimit(ctx, req.RegID); err != nil {
+		return nil, err
 	}
 
-	if err := ra.checkInvalidAuthorizationLimit(ctx, regID, identifier.Value); err != nil {
-		return core.Authorization{}, err
+	if err := ra.checkInvalidAuthorizationLimit(ctx, req.RegID, acmeIdentifier.Value); err != nil {
+		return nil, err
 	}
 
 	if ra.reuseValidAuthz {
 		now := ra.clk.Now().UnixNano()
 		authzMapPB, err := ra.SA.GetValidAuthorizations2(ctx, &sapb.GetValidAuthorizationsRequest{
-			RegistrationID: regID,
-			Domains:        []string{identifier.Value},
+			RegistrationID: req.RegID,
+			Domains:        []string{acmeIdentifier.Value},
 			Now:            now,
 		})
 		if err != nil {
 			outErr := berrors.InternalServerError(
-				"unable to get existing validations for regID: %d, identifier: %s, %s",
-				regID,
-				identifier.Value,
+				"unable to get existing validations for request.RegID: %d, acmeIdentifier: %s, %s",
+				req.RegID,
+				acmeIdentifier.Value,
 				err,
 			)
 			ra.log.Warning(outErr.Error())
-			return core.Authorization{}, outErr
+			return nil, outErr
 		}
 		auths, err := bgrpc.PBToAuthzMap(authzMapPB)
 		if err != nil {
-			return core.Authorization{}, err
+			return nil, err
 		}
 
-		if existingAuthz, ok := auths[identifier.Value]; ok {
+		if existingAuthz, ok := auths[acmeIdentifier.Value]; ok {
 			if ra.authzValidChallengeEnabled(existingAuthz) {
 				// The existing authorization must not expire within the next 24 hours for
 				// it to be OK for reuse
 				reuseCutOff := ra.clk.Now().Add(time.Hour * 24)
 				if existingAuthz.Expires.After(reuseCutOff) {
 					ra.reusedValidAuthzCounter.Inc()
-					return *existingAuthz, nil
+					return bgrpc.AuthzToPB(*existingAuthz)
 				}
 			}
 		}
 	}
 
-	identifierTypeString := string(identifier.Type)
-	req := &sapb.GetPendingAuthorizationRequest{
-		RegistrationID:  regID,
+	identifierTypeString := string(acmeIdentifier.Type)
+	saReq := &sapb.GetPendingAuthorizationRequest{
+		RegistrationID:  req.RegID,
 		IdentifierType:  identifierTypeString,
-		IdentifierValue: identifier.Value,
+		IdentifierValue: acmeIdentifier.Value,
 		ValidUntil:      ra.clk.Now().Add(time.Hour).UnixNano(),
 	}
-	pendingPB, err := ra.SA.GetPendingAuthorization2(ctx, req)
+	pendingPB, err := ra.SA.GetPendingAuthorization2(ctx, saReq)
```

Comment:
This _could_ be inlined, since this `saReq` is only used right here, but up to you.

Reply:
done